core/corehttp/gateway_handler.go

```diff
@@ -383,6 +383,15 @@ func (i *gatewayHandler) serveFile(w http.ResponseWriter, req *http.Request, nam
 		}
 	}
 
+	buf := make([]byte, 512)
+	_, _ = content.Read(buf)
+	mime := http.DetectContentType(buf)
+	
+	if strings.HasPrefix(mime, "text/html;") {
+		mime = "text/html"
+	}
+	w.Header().Set("Content-Type", mime)
```

Comment:
The file type detector inside `ServeContent` also uses the extension. We should probably do that as well.

(although ignoring the extension when there's a `?` (query) present may make sense).